src/sql.rs

```diff
@@ -137,46 +137,49 @@ impl Sql {
             .to_str()
             .with_context(|| format!("path {path:?} is not valid unicode"))?
             .to_string();
-        let res = self
-            .call_write(move |conn| {
-                // Check that backup passphrase is correct before resetting our database.
-                conn.execute("ATTACH DATABASE ? AS backup KEY ?", (path_str, passphrase))
-                    .context("failed to attach backup database")?;
-                if let Err(err) = conn
-                    .query_row("SELECT count(*) FROM sqlite_master", [], |_row| Ok(()))
-                    .context("backup passphrase is not correct")
-                {
-                    conn.execute("DETACH DATABASE backup", [])
-                        .context("failed to detach backup database")?;
-                    return Err(err);
-                }
 
-                // Reset the database without reopening it. We don't want to reopen the database because we
-                // don't have main database passphrase at this point.
-                // See <https://sqlite.org/c3ref/c_dbconfig_enable_fkey.html> for documentation.
-                // Without resetting import may fail due to existing tables.
+        // The config cache is wrong now that we have a different database
+        let mut config_cache = self.config_cache.write().await;
```

Comment:
Keep `config_cache` locked all the time the db is imported so that nobody can use invalid values from there. And clear it immediately so as not to forget

Reply:
Maybe write this comment directly in the code :)